### DISCUSSION TITLE
Specifying times in month view seem redundant when week view exists and how to combine both

### DISCUSSION THREAD

**Discussion by kleenkanteen** [lukevella/rallly#779] [Ideas]
One can specify the times for each day using month view:

![image](https://github.com/lukevella/rallly/assets/67878058/414aa2a6-6e7a-46fd-8b15-c735b3cd286d)

Week view also exists though, and I find it a lot easier to select different times for each day as compared to month view. In month view you have to click each time slot, scroll to the desired time, for each day. It's cumbersome. We could incorporate week view as a popup into the "Specify times" option in the month view.

Second suggestion, like when2meet.com, it would be nice if by default you set a bulk start and end time for all selected days. Right now you can apply one day's times to all other days, but it's hidden behind an option: 

![image](https://github.com/lukevella/rallly/assets/67878058/2cd14ad5-86a4-4fed-aa93-d85732ee7e43)

When creating an event, when2meet.com keeps it simple and doesn't let you break up times for each day, all days must start and stop at the same time. I think this should be the default. So when you click "Specify times", it should have just 1 day's slot show that automatically apply to all days selected in the month. If a user wants to select more granular time slots, below they can select "select times separately" which makes the week view pop up. Simple UI at first, more complex as needed. 

This idea sounds good?

**Comment by lukevella** [comment]
> Week view also exists though, and I find it a lot easier to select different times for each day as compared to month view. In month view you have to click each time slot, scroll to the desired time, for each day. It's cumbersome. We could incorporate week view as a popup into the "Specify times" option in the month view.

Maybe. Depends heavily on the execution how good of an idea this is. Needs to be fleshed out a bit more with some actual designs showing what the UX would be like. One thing to keep in mind is that the clicking and dragging timeslots doesn't work so great on mobile devices which is why I made month view the default.

> Second suggestion, like when2meet.com, it would be nice if by default you set a bulk start and end time for all selected days. Right now you can apply one day's times to all other days, but it's hidden behind an option:

[Have a look here](https://rallly.co/blog/the-future-of-rallly) for more details about long-term plans for Rallly. I do plan on offering a similar solution to when2meet (i.e. availability grids or heatmaps) but this is a different solution to meeting polls. Meeting polls get less useful the more options there are so enabling users to add more options quickly is not the right direction.

**Comment by kleenkanteen** [comment]
Any analytics on how many users make polls on computer vs. phone? Could use this to prioritize how much features should be mobile friendly.

> One thing to keep in mind is that the clicking and dragging timeslots doesn't work so great on mobile devices which is why I made month view the default.

Seems so, I just tried https://web.dev/drag-and-drop/ on my phone and laptop and it did not work on my phone in the demo they gave. I just tried on my phone to use drag and drop to fill out my availablity on a when2meet.com poll and it worked well. Only issue was you couldn't select just one 15 minute incement to select or unselect it. It would change color then revert back for some reason.

For desktop, putting the week view selection under the month view "Specify times" option would be straightforward. Mobile is trickier and definitely needs a design overhaul as well looking at this:

![image](https://github.com/lukevella/rallly/assets/67878058/1a3a7f5a-486b-4713-b2fb-ed3335ec81bc)

To prototype my idea's UI design should I use Figma or write code and share the PR? First time I will do something like this.

> Meeting polls get less useful the more options there are so enabling users to add more options quickly is not the right direction.

I mean selecting times when creating a poll. If you are talking about poll creation, I don't see why this is the case. Many times when I used when2meet it was a certain time frame that the creator wanted the meeting, say Mon-Fri, 5-8pm. It let you select 5-8pm as the possible timings for all days this way, look at the right side, under "What times might work?":

![image](https://github.com/lukevella/rallly/assets/67878058/b2dfe1e0-065a-499b-9294-abafb2773221)

If one was to use rally to create a poll with the timings of Mon-Fri 5-8pm, compared to when2meet it would require several more clicks with week view or a lot more clicks with month view. Having a bulk start and end time for all days selected simplifies things. What do you think?
